pandas/core/arrays/base.py

```diff
@@ -2352,6 +2352,9 @@ def _groupby_op(
         # GH#43682
         if isinstance(self.dtype, StringDtype):
             # StringArray
+            if op.how not in ["any", "all"]:
+                # Fail early to avoid conversion to object
+                op._get_cython_function(op.kind, op.how, np.dtype(object), False)
```

Comment:
Does this fail prior to any computation? I.e. is O(1)

Reply:
No, this only does some dtype related checks, e.g. it is O(1)